internal/handler/acccess_check_test.go

```diff
@@ -0,0 +1,76 @@
+package handler
```

Comment:
tests should be in the file named with _test suffix as the file originating function. auth_test.go in this case. There is no reason to have a separate filename.